src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -44,6 +45,12 @@ public TelemetryClient(TelemetryConfiguration configuration)
             }
 
             this.configuration = configuration;
+
+            // If someone created configuration from scratch and forgot to initialize the channel then use the default.
+            if (this.configuration.TelemetryChannel == null)
+            {
+                throw new ArgumentException("The specified configuration does not have a telemetry channel.", "configuration");
```

Comment:
Would our customers prefer an `ArgumentException` with an actionable message here or a `NullReferenceException`, deep in the call to the `TelemetryClient.Track(ITelemetry)?